File: fix_urls.js
const fs = require('fs');
let code = fs.readFileSync('d:/sasProject/Web-main/src/AdminDashboard.jsx', 'utf8');

code = code.replace(/\`\$\{baseUrl\}([^']*?)'/g, '\`${baseUrl}$1\`');
code = code.replace(/\`\$\{baseUrl\}([^"]*?)"/g, '\`${baseUrl}$1\`');
code = code.replace(/['"]http:\/\/localhost:3000([^'"]*)['"]/g, '\`${baseUrl}$1\`');

// Let's also fix the ones from lines 737, 759, 787 that were like fetch(`/api/...`) but should be `${baseUrl}/api/...`
code = code.replace(/fetch\(['"]\/api\//g, 'fetch(`${baseUrl}/api/');

fs.writeFileSync('d:/sasProject/Web-main/src/AdminDashboard.jsx', code);
console.log('Fixed quotes');

File: fix_urls.py
import os

filepath = 'd:/sasProject/Web-main/src/AdminDashboard.jsx'
with open(filepath, 'r', encoding='utf-8') as f:
    code = f.read()

import_end = code.find('\nconst AdminDashboard =')
api_base_def = '\nconst API_BASE = import.meta.env.VITE_API_URL || "http://localhost:3000";\nconst baseUrl = API_BASE.replace(/\\/$/, "");\n'
code = code[:import_end] + api_base_def + code[import_end:]

code = code.replace("'http://localhost:3000", "`${baseUrl}")
code = code.replace("http://localhost:3000'", "${baseUrl}`")
code = code.replace('"http://localhost:3000', '`${baseUrl}')
code = code.replace('http://localhost:3000"', '${baseUrl}`')
code = code.replace('`http://localhost:3000', '`${baseUrl}')
code = code.replace('const baseUrl = API_BASE.replace(/\\/$/, "");', '')
# put one declaration back
code = code.replace('const API_BASE = import.meta.env.VITE_API_URL || "http://localhost:3000";\n', 'const API_BASE = import.meta.env.VITE_API_URL || "http://localhost:3000";\nconst baseUrl = API_BASE.replace(/\\/$/, "");\n', 1)

with open(filepath, 'w', encoding='utf-8') as f:
    f.write(code)

print("Done")
